lib/metrics/statsd.js

```diff
@@ -0,0 +1,73 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var StatsD = require('node-statsd')
+
+var STATSD_PREFIX = 'fxa.auth.'
+
+function messageSentCallback (err) {
+  // this only gets called once after all messages have been sent
+  if (err) {
+    console.log('StatsD error:', err)
+  }
+}
+
+function StatsDCollector() {
+  var config = require('../../config')
+  var statsdConfig = config.get('statsd')
+
+  this.host = statsdConfig.host
+  this.port = statsdConfig.port
+  this.sampleRate = statsdConfig.sample_rate
+  this.connected = false
+  this.client = null
+}
+
+StatsDCollector.prototype = {
+  /**
+   * Initializes a StatsD socket client
+   */
+  init: function () {
+    var client = this.client = new StatsD(this.host, this.port)
+
+    if (client.socket) {
+      this.connected = true
+      client.socket.on('error', function (error) {
+        console.log('Error in stats socket: ', error)
```

Comment:
> Should we take some action here to e.g. tear down and reconnect to the server?

A few things here, it is a UDP socket so there is no "reconnect". In my experience with the content-server this would only happen on start up if the agent is not working. 

I want to remove the `console.log` and put a proper log though...